lib/core/src/sdk.rs

```diff
@@ -95,11 +102,33 @@ impl LiquidSdk {
         Ok(sdk)
     }
 
+    fn validate_api_key(api_key: &str) -> Result<()> {
+        let api_key_decoded = base64::engine::general_purpose::STANDARD
+            .decode(api_key.as_bytes())
+            .map_err(|err| anyhow!("Could not base64 decode the Breez API key: {err:?}"))?;
+        let (_rem, cert) = parse_x509_certificate(&api_key_decoded)
+            .map_err(|err| anyhow!("Invaid certificate for Breez API key: {err:?}"))?;
+
+        match cert.public_key().parsed() {
+            Ok(pk) => ensure_sdk!(
+                pk == BREEZ_CA_PUBLIC_KEY,
```

Comment:
My concern here is that from some reason we will have to replace our. CA and sign new certificates with a different CA. This will break existing clients.
Since this validation is not for security reasons but only to ensure the user didn't do a mistake and used the wrong key so perhaps we can just verify the CA name starts with "Breez". Ii talked to @yaslama and this will not change also on future certificates.